core/src/components/datetime/datetime.tsx

```diff
@@ -1474,7 +1581,7 @@ export class Datetime implements ComponentInterface {
           });
 
           this.setActiveParts({
-            ...this.activeParts,
+            ...(this.activeParts as DatetimeParts),
```

Comment:
Similar thoughts here around type narrowing. If at some point a feature sets `activeParts` to an array by accident, type checking will not catch the error since we are type casting.